Package request: arma3-unix-launcher

**Project description**
A clean, intuitive Arma 3 + DayZ SA Launcher for GNU/Linux and MacOS.

**Metadata**

* homepage URL: [https://github.com/muttleyxd/arma3-unix-launcher/](https://github.com/muttleyxd/arma3-unix-launcher/)
* source URL: [https://github.com/muttleyxd/arma3-unix-launcher/](https://github.com/muttleyxd/arma3-unix-launcher/)
* license: mit (does require the steamworks SDK which has its own license.)
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Looks great to me! Run it through `nixfmt-rfc-style` to pretty it up and I think it'd fit in just fine in nixpkgs. I am getting a build error with the DayZ launcher, though:

```
[ 96%] Building CXX object src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/dayz-linux-launcher_autogen/EWIEGA46WW/qrc_stylesheet.cpp.o
/build/source/src/dayz-linux-launcher/mainwindow.cpp: In constructor 'MainWindow::MainWindow(std::unique_ptr<DayZ::Client>, const std::filesystem::__cxx11::path&, bool, QWidget*)':
/build/source/src/dayz-linux-launcher/mainwindow.cpp:50:29: warning: unused parameter 'use_steam_integration' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-parameter-Wunused-parameter8;;]
   50 |                        bool use_steam_integration, QWidget *parent) :
      |                        ~~~~~^~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/format.h:49,
                 from /build/source/src/arma3-unix-launcher-library/./mod.hpp:7,
                 from /build/source/src/dayz-linux-launcher-library/./dayzclient.hpp:8,
                 from /build/source/src/dayz-linux-launcher/mainwindow.h:16,
                 from /build/source/src/dayz-linux-launcher/mainwindow.cpp:2:
/nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h: In instantiation of 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; T = const nlohmann::json_abi_v3_11_3::basic_json<>; typename std::enable_if<PACKED, int>::type <anonymous> = 0]':
/nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1842:51:   required from 'constexpr fmt::v10::format_arg_store<Context, Args>::format_arg_store(T& ...) [with T = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>, const char*}; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; Args = {nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>, const char*}]'
/nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1860:18:   required from 'constexpr fmt::v10::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<T>::type>::type ...> fmt::v10::make_format_args(T& ...) [with Context = basic_format_context<appender, char>; T = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>, const char*}]'
/nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:328:75:   required from 'void spdlog::logger::log_(spdlog::source_loc, spdlog::level::level_enum, spdlog::string_view_t, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; spdlog::string_view_t = fmt::v10::basic_string_view<char>]'
/nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:80:13:   required from 'void spdlog::logger::log(spdlog::source_loc, spdlog::level::level_enum, fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
/nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:85:12:   required from 'void spdlog::logger::log(spdlog::level::level_enum, fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
/nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:145:12:   required from 'void spdlog::logger::warn(fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
/nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/spdlog.h:173:31:   required from 'void spdlog::warn(fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
/build/source/src/dayz-linux-launcher/mainwindow.cpp:719:25:   required from here
/nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1600:63: error: 'fmt::v10::detail::type_is_unformattable_for<const nlohmann::json_abi_v3_11_3::basic_json<>, char> _' has incomplete type
 1600 |     type_is_unformattable_for<T, typename Context::char_type> _;
      |                                                               ^
/nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1604:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1604 |       formattable,
      |       ^~~~~~~~~~~
/nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1604:7: note: 'formattable' evaluates to false
make[2]: *** [src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/build.make:189: src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/mainwindow.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:465: src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/1vpamslg482b1nzzz4vcj2w82ankadxf-arma3-unix-launcher-git.drv' failed with exit code 2;
       last 25 log lines:
       > In file included from /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/format.h:49,
       >                  from /build/source/src/arma3-unix-launcher-library/./mod.hpp:7,
       >                  from /build/source/src/dayz-linux-launcher-library/./dayzclient.hpp:8,
       >                  from /build/source/src/dayz-linux-launcher/mainwindow.h:16,
       >                  from /build/source/src/dayz-linux-launcher/mainwindow.cpp:2:
       > /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h: In instantiation of 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; T = const nlohmann::json_abi_v3_11_3::basic_json<>; typename std::enable_if<PACKED, int>::type <anonymous> = 0]':
       > /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1842:51:   required from 'constexpr fmt::v10::format_arg_store<Context, Args>::format_arg_store(T& ...) [with T = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>, const char*}; Context = fmt::v10::basic_format_context<fmt::v10::appender, char>; Args = {nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>, const char*}]'
       > /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1860:18:   required from 'constexpr fmt::v10::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<T>::type>::type ...> fmt::v10::make_format_args(T& ...) [with Context = basic_format_context<appender, char>; T = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>, const char*}]'
       > /nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:328:75:   required from 'void spdlog::logger::log_(spdlog::source_loc, spdlog::level::level_enum, spdlog::string_view_t, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; spdlog::string_view_t = fmt::v10::basic_string_view<char>]'
       > /nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:80:13:   required from 'void spdlog::logger::log(spdlog::source_loc, spdlog::level::level_enum, fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
       > /nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:85:12:   required from 'void spdlog::logger::log(spdlog::level::level_enum, fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
       > /nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/logger.h:145:12:   required from 'void spdlog::logger::warn(fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
       > /nix/store/r7f301cdb53hbsihbmivfq0zwz2vm3pr-spdlog-1.13.0-dev/include/spdlog/spdlog.h:173:31:   required from 'void spdlog::warn(fmt::v10::format_string<T ...>, Args&& ...) [with Args = {const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*}; fmt::v10::format_string<T ...> = fmt::v10::basic_format_string<char, const nlohmann::json_abi_v3_11_3::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::json_abi_v3_11_3::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> >, void>&, const char*>]'
       > /build/source/src/dayz-linux-launcher/mainwindow.cpp:719:25:   required from here
       > /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1600:63: error: 'fmt::v10::detail::type_is_unformattable_for<const nlohmann::json_abi_v3_11_3::basic_json<>, char> _' has incomplete type
       >  1600 |     type_is_unformattable_for<T, typename Context::char_type> _;
       >       |                                                               ^
       > /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1604:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
       >  1604 |       formattable,
       >       |       ^~~~~~~~~~~
       > /nix/store/fr5vsxz2py75m42kj1r3fqgc2ll7pjwa-fmt-10.2.1-dev/include/fmt/core.h:1604:7: note: 'formattable' evaluates to false
       > make[2]: *** [src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/build.make:189: src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/mainwindow.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:465: src/dayz-linux-launcher/CMakeFiles/dayz-linux-launcher.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/1vpamslg482b1nzzz4vcj2w82ankadxf-arma3-unix-launcher-git.drv'.
```

This might inspire me to go back to arma 3. Longtime OFP/arma fan!

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -5436,6 +5436,12 @@
     githubId = 252042;
     keys = [ { fingerprint = "85F3 72DF 4AF3 EF13 ED34  72A3 0AAF 2901 E804 0715"; } ];
   };
+  DrymarchonShaun = {
+    name = "Shaun";
+    email = "drymarchonshaun@protonmail.com";
+    github = "DrymarchonShaun";
+    githubId = 40149778;
+  };
   dsalaza4 = {
     email = "podany270895@gmail.com";
     github = "dsalaza4";

--- a/pkgs/by-name/ar/arma3-unix-launcher/disable_steam_integration.patch
+++ b/pkgs/by-name/ar/arma3-unix-launcher/disable_steam_integration.patch
@@ -1,0 +1,25 @@
+diff --git a/src/arma3-unix-launcher/mainwindow.cpp b/src/arma3-unix-launcher/mainwindow.cpp
+index 66b73cc..f89f66b 100644
+--- a/src/arma3-unix-launcher/mainwindow.cpp
++++ b/src/arma3-unix-launcher/mainwindow.cpp
+@@ -56,6 +56,3 @@ MainWindow::MainWindow(std::unique_ptr<ARMA3::Client> arma3_client, std::filesys
+ {
+-    if (use_steam_integration)
+-        steam_integration = std::make_unique<Steam::SteamIntegration>(ARMA3::Definitions::app_id);
+-    else
+-        steam_integration = std::make_unique<Steam::IntegrationStub>(ARMA3::Definitions::app_id);
++    steam_integration = std::make_unique<Steam::IntegrationStub>(ARMA3::Definitions::app_id);
+ 
+diff --git a/src/dayz-linux-launcher/mainwindow.cpp b/src/dayz-linux-launcher/mainwindow.cpp
+index d9223db..5773593 100644
+--- a/src/dayz-linux-launcher/mainwindow.cpp
++++ b/src/dayz-linux-launcher/mainwindow.cpp
+@@ -56,6 +56,3 @@ MainWindow::MainWindow(std::unique_ptr<DayZ::Client> arma3_client, std::filesyst
+ {
+-    if (use_steam_integration)
+-        steam_integration = std::make_unique<Steam::SteamIntegration>(DayZ::Definitions::app_id);
+-    else
+-        steam_integration = std::make_unique<Steam::IntegrationStub>(DayZ::Definitions::app_id);
++    steam_integration = std::make_unique<Steam::IntegrationStub>(DayZ::Definitions::app_id);
+ 
+

--- a/pkgs/by-name/ar/arma3-unix-launcher/dont_fetch_dependencies.patch
+++ b/pkgs/by-name/ar/arma3-unix-launcher/dont_fetch_dependencies.patch
@@ -1,0 +1,115 @@
+diff --git a/cmake/external_dependencies.cmake b/cmake/external_dependencies.cmake
+index 2eb6ec5..9f1d67e 100644
+--- a/cmake/external_dependencies.cmake
++++ b/cmake/external_dependencies.cmake
+@@ -4,7 +4,7 @@ include(FetchContent)
+ 
+ function(setup_library SOURCE_TO_TEST)
+     set(boolArgs HEADER_ONLY)
+-    set(oneValueArgs NAME CXX_FLAGS GIT_REPOSITORY GIT_TAG TEST_DEFINITIONS TEST_LINK_LIBS)
++    set(oneValueArgs NAME CXX_FLAGS URL GIT_TAG TEST_DEFINITIONS TEST_LINK_LIBS)
+     set(multiValueArgs WHEN)
+     cmake_parse_arguments(LIB_SETUP "${boolArgs}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+ 
+@@ -27,12 +27,11 @@ function(setup_library SOURCE_TO_TEST)
+     endif()
+ 
+     FetchContent_Declare(${LIB_SETUP_NAME}
+-                         GIT_REPOSITORY ${LIB_SETUP_GIT_REPOSITORY}
+-                         GIT_TAG ${LIB_SETUP_GIT_TAG})
++        URL ${LIB_SETUP_URL})
+     FetchContent_GetProperties(${LIB_SETUP_NAME})
+     set(POPULATED "${LIB_SETUP_NAME}_POPULATED")
+     if (NOT "${POPULATED}")
+-        message("-- Downloading ${LIB_SETUP_NAME} from ${LIB_SETUP_GIT_REPOSITORY}")
++        message("-- Downloading ${LIB_SETUP_NAME} from ${LIB_SETUP_URL}")
+         FetchContent_Populate(${LIB_SETUP_NAME})
+         set(SRCDIR "${LIB_SETUP_NAME}_SOURCE_DIR")
+         set(BINDIR "${LIB_SETUP_NAME}_BINARY_DIR")
+@@ -51,8 +50,7 @@ function(setup_argparse)
+             }")
+     setup_library("${CHECK_SOURCE}"
+                   NAME argparse
+-                  GIT_REPOSITORY https://github.com/p-ranav/argparse.git
+-                  GIT_TAG 45664c4
++                  URL @argparse_src@
+                   HEADER_ONLY
+                   )
+     if (NOT TARGET argparse::argparse)
+@@ -64,7 +62,7 @@ function(setup_curlpp)
+     set(CHECK_SOURCE "#error unimplemented}")
+     setup_library("${CHECK_SOURCE}"
+                   NAME curlpp
+-                  GIT_REPOSITORY https://github.com/jpbarrette/curlpp.git
++                  URL @curlpp_src@
+                   )
+     if (NOT APPLE)
+         set(CURLPP_LIB_PATH1 "${curlpp_BINARY_DIR}/libcurlpp.so" PARENT_SCOPE)
+@@ -79,7 +77,7 @@ function(setup_doctest)
+                       #include <doctest/doctest.h>")
+     setup_library("${CHECK_SOURCE}"
+                   NAME doctest
+-                  GIT_REPOSITORY https://github.com/onqtam/doctest.git
++                  URL @doctest_src@
+                   HEADER_ONLY
+                   )
+     add_library(doctest::doctest ALIAS doctest)
+@@ -100,8 +98,7 @@ function(setup_fmt)
+         }")
+     setup_library("${CHECK_SOURCE}"
+                   NAME fmt
+-                  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+-                  GIT_TAG 8.1.1
++                  URL @fmt_src@
+                   TEST_LINK_LIBS fmt
+                   )
+ 
+@@ -126,7 +123,7 @@ function(setup_nlohmann_json)
+         add_library(nlohmann_json INTERFACE)
+     else()
+         FetchContent_Declare(nlohmann_json
+-                             URL https://github.com/nlohmann/json/releases/download/v3.7.3/include.zip)
++                             URL @nlohmann_json_src@)
+         FetchContent_GetProperties(nlohmann_json)
+         if (NOT nlohmann_json_POPULATED)
+             FetchContent_Populate(nlohmann_json)
+@@ -146,8 +143,7 @@ function(setup_pugixml)
+         }")
+     setup_library("${CHECK_SOURCE}"
+                   NAME pugixml
+-                  GIT_REPOSITORY https://github.com/muttleyxd/pugixml.git
+-                  GIT_TAG simple-build-for-a3ul
++                  URL @pugixml_src@
+                   TEST_LINK_LIBS pugixml
+                   )
+     get_target_property(TARGET_TYPE pugixml TYPE)
+@@ -188,8 +184,7 @@ function(setup_spdlog)
+ 
+     setup_library("${CHECK_SOURCE}"
+                   NAME spdlog
+-                  GIT_REPOSITORY https://github.com/gabime/spdlog.git
+-                  GIT_TAG v1.x
++                  URL @spdlog_src@
+                   TEST_DEFINITIONS -DSPDLOG_FMT_EXTERNAL
+                   TEST_LINK_LIBS ${FMT_TARGET_NAME}
+                   CXX_FLAGS "-Wno-attributes -Wno-reorder -Wno-redundant-move"
+@@ -198,8 +193,7 @@ endfunction()
+ 
+ function(setup_steamworkssdk)
+     FetchContent_Declare(steamworkssdk
+-        URL https://github.com/julianxhokaxhiu/SteamworksSDKCI/releases/download/1.53/SteamworksSDK-v1.53.0_x64.zip
+-        URL_HASH MD5=322c2c90c3ab76201c92f4a2c443f664
++        URL @steamworkssdk_src@
+         CONFIGURE_COMMAND ""
+         BUILD_COMMAND ""
+         )
+@@ -241,8 +235,7 @@ function(setup_trompeloeil)
+                       int main() { }")
+     setup_library("${CHECK_SOURCE}"
+                   NAME trompeloeil
+-                  GIT_REPOSITORY https://github.com/rollbear/trompeloeil.git
+-                  GIT_TAG 64fd171
++                  URL @trompeloeil_src@
+                   HEADER_ONLY
+                   )
+     add_library(trompeloeil::trompeloeil ALIAS trompeloeil)

--- a/pkgs/by-name/ar/arma3-unix-launcher/package.nix
+++ b/pkgs/by-name/ar/arma3-unix-launcher/package.nix
@@ -1,0 +1,88 @@
+{
+  lib,
+  stdenv,
+  cmake,
+  curl,
+  curlpp,
+  doctest,
+  fetchFromGitHub,
+  fetchurl,
+  fmt,
+  nlohmann_json,
+  qt5,
+  spdlog,
+  substituteAll,
+  trompeloeil,
+  buildDayZLauncher ? false,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "arma3-unix-launcher";
+  version = "413";
+
+  src = fetchFromGitHub {
+    owner = "muttleyxd";
+    repo = "arma3-unix-launcher";
+    rev = "2ea62d961522f1542d4c8e669ef5fe856916f9ec";
+    hash = "sha256-uym93mYmVj9UxT8RbwdRUyIPrQX7nZTNWUUVjxCQmVU=";
+  };
+
+  patches = [
+    # prevent CMake from trying to get libraries on the internet
+    (substituteAll {
+      src = ./dont_fetch_dependencies.patch;
+      argparse_src = fetchFromGitHub {
+        owner = "p-ranav";
+        repo = "argparse";
+        rev = "45664c4e9f05ff287731a9ff8b724d0c89fb6e77";
+        sha256 = "sha256-qLD9zD6hbItDn6ZHHWBXrAWhySvqcs40xA5+C/5Fkhw=";
+      };
+      curlpp_src = curlpp.src;
+      doctest_src = doctest;
+      fmt_src = fmt;
+      nlohmann_json_src = nlohmann_json;
+      pugixml_src = fetchFromGitHub {
+        owner = "muttleyxd";
+        repo = "pugixml";
+        rev = "simple-build-for-a3ul";
+        sha256 = "sha256-FpREdz6DbhnLDGOuQY9rU17SSd6ngA4WfO0kGHqGJPM=";
+      };
+      spdlog_src = spdlog;
+      steamworkssdk_src = fetchurl {
+        url = "https://github.com/julianxhokaxhiu/SteamworksSDKCI/releases/download/1.53/SteamworksSDK-v1.53.0_x64.zip";
+        sha256 = "sha256-6PQGaPsaxBg/MHVWw2ynYW6LaNSrE9Rd9Q9ZLKFGPFA=";
+      };
+      trompeloeil_src = trompeloeil;
+    })
+    # game won't launch with steam integration anyways, disable it
+    ./disable_steam_integration.patch
+  ];
+
+  nativeBuildInputs = [
+    qt5.wrapQtAppsHook
+    cmake
+  ];
+
+  buildInputs = [
+    spdlog
+    curlpp.src
+    curl
+    qt5.qtbase
+    qt5.qtsvg
+  ];
+
+  cmakeFlags = [ "-Wno-dev" ] ++ lib.optionals buildDayZLauncher [ "-DBUILD_DAYZ_LAUNCHER=ON" ];
+
+  meta = {
+    homepage = "https://github.com/muttleyxd/arma3-unix-launcher/";
+    description = "Clean, intuitive Arma 3 + DayZ SA Launcher";
+    license = with lib.licenses; [
+      # Launcher
+      mit
+      # Steamworks SDK
+      unfree
+    ];
+    maintainers = with lib.maintainers; [ DrymarchonShaun ];
+    mainProgram = "arma3-unix-launcher";
+    platforms = with lib.platforms; linux ++ darwin;
+  };
+})

